clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1629,37 +1630,90 @@ void FunctionVariableConstraint::brainTransplant(ConstraintVariable *FromCV,
   auto RetVar = getOnly(returnVars);
   RetVar->brainTransplant(FromRetVar, I);
   // Transplant params.
-  assert(From->numParams() == numParams());
-  for (unsigned i = 0; i < From->numParams(); i++) {
-    CVarSet &FromP = From->getParamVar(i);
-    CVarSet &P = getParamVar(i);
-    auto FromVar = getOnly(FromP);
-    auto Var = getOnly(P);
-    Var->brainTransplant(FromVar, I);
+  if (numParams() == From->numParams()) {
+    for (unsigned i = 0; i < From->numParams(); i++) {
+      CVarSet &FromP = From->getParamVar(i);
+      CVarSet &P = getParamVar(i);
+      auto FromVar = getOnly(FromP);
+      auto Var = getOnly(P);
+      Var->brainTransplant(FromVar, I);
+    }
+  } else if (numParams() != 0 && From->numParams() == 0) {
+    auto &CS = I.getConstraints();
+    vector<ParamDeferment> defers = From->getDeferredParams();
+    defers.insert(defers.end(),
+                  getDeferredParams().begin(), getDeferredParams().end());
```

Comment:
I don't follow what this is doing. Why are you not just iterating over `defers` directly? What is this `insert` doing?

Reply:
Put another way: For `brainTransplant` there should be no deferred params for the definition. I.e., add an assert that `  getDeferredParams() == 0` if you like.

Reply:
Combining the defers in `this` and `From` into one vector.

Reply:
See my second point: There should be no such parms in `this`. We are only calling `brainTransplant` on a definition that we are seeing for the first time.